### DISCUSSION TITLE
Get Entities - Doesn't see all entities?

### DISCUSSION THREAD

**Discussion by Smith754** [zachowj/node-red-contrib-home-assistant-websocket#428] [Q&A]
I have a simple Get Entities node but it doesn't show anything for "attribute.battery_level" even though I have over a dozen battery powered motion detector entities with this. It used to work, but now it doesn't.  Is it because I switched to Z-Wave JS instead of Z-Wave (deprecated) within Home Assistant?  Or, am I just doing it wrong because the node has been updated?

Under server for this node I have "Cache Autocomplete Results" unchecked, but it doesn't seem to help. 

`[{"id":"b910f692.c36ee8","type":"inject","z":"76ff7f9f.21faf","name":"","repeat":"","crontab":"00 20 * * *","once":false,"onceDelay":"8","topic":"","payload":"","payloadType":"date","x":150,"y":1540,"wires":[["337ced06.58de82"]]},{"id":"337ced06.58de82","type":"ha-get-entities","z":"76ff7f9f.21faf","name":"","server":"66d9e619.115338","version":0,"rules":[{"property":"attributes.battery_level","logic":"gt","value":"0","valueType":"num"}],"output_type":"array","output_empty_results":false,"output_location_type":"msg","output_location":"payload","output_results_count":1,"x":310,"y":1540,"wires":[["1d0ce920.42dc57"]]},{"id":"1d0ce920.42dc57","type":"debug","z":"76ff7f9f.21faf","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":450,"y":1540,"wires":[]},{"id":"66d9e619.115338","type":"server","name":"Home Assistant","version":1,"legacy":false,"addon":true,"rejectUnauthorizedCerts":true,"ha_boolean":"y|yes|true|on|home|open","connectionDelay":true,"cacheJson":false}]`

**Comment by zachowj** [comment]
You can easily check if there are entities with the attribute using the `developer tools` > `state` and typing in `battery_level` into the attributes column.

**Reply by Smith754** [reply]
I see no way to type "battery_level" into the attributes column on the Developer->States page. 
There is a dropdown list of entities. A state entry box. And a state attributes box.  The only buttons are Set State, or a circular arrow.
If I type "battry_level" into state attributes then you can't use the Set state button because it wants an enitity name in Entity, and the circular arrow just erases what's in the state attributes box.

I can scroll down and find:
**state_class: measurement
unit_of_measurement: %
friendly_name: Motion Den: Battery Level
device_class: battery**

**Reply by zachowj** [reply]
![image](https://user-images.githubusercontent.com/37859597/121796381-5fbeb500-cbcd-11eb-87ff-656793684a52.png)

**Reply by Smith754** [reply]
You meant literally into the top of the column, I see.  (Not drunk, just stupid, LOL. Sorry).

Well, I get the same list with "battery", or "battery level". About a dozen or more entities. So I know they exist.
But the poor little Get Entities node doesn't list anything with the word "battery" in it.  I thought the Property box was supposed to have a list of everything. It just returns "No Results".

**Reply by zachowj** [reply]
Do you get any entities that have `battery_level` on the left side of the colon?

![image](https://user-images.githubusercontent.com/37859597/121797328-669cf600-cbd4-11eb-8998-9f709e5b2dcc.png)

**Comment by Smith754** [comment]
Thanks for all the info zachowj!  Figured it out.  Things had changed a little in HA and node-red. 
In case anyone else reads this just create 2 items with the following properties in the Get Entities node to get a list of all entities with a battery state (%):

```
Property:  attributes.device_class
is: [a/z] Battery
Property: state
>: [09] 0
```
